annotation-processor/src/main/java/org/fulib/fx/ControllerAnnotationProcessor.java

```diff
@@ -277,4 +288,90 @@ private boolean isProvider(TypeMirror typeMirror) {
         return typeMirror.toString().startsWith("javax.inject.Provider");
     }
 
+    /**
+     * Checks if the given method overrides another event method.
+     *
+     * @param method The method to check
+     */
+    private void checkOverrides(ExecutableElement method) {
+        TypeMirror clazz = method.getEnclosingElement().asType();
+        TypeElement element = (TypeElement) processingEnv.getTypeUtils().asElement(clazz);
+        TypeMirror parentClazz = element.getSuperclass();
+
+        // If no parent class is found, the method cannot override anything
+        if (parentClazz.getKind() == TypeKind.NONE) {
+            return;
+        }
+
+        TypeElement parentElement = (TypeElement) processingEnv.getTypeUtils().asElement(parentClazz);
+
+        streamAllMethods(parentElement)
+                .filter(otherMethod -> otherMethod.getSimpleName().equals(method.getSimpleName()))
+                .filter(this::isEventMethod)
+                .filter(otherMethod -> processingEnv.getTypeUtils().isSubsignature((ExecutableType) method.asType(), (ExecutableType) otherMethod.asType()))
+                .findFirst()
+                .ifPresent(overriddenMethod -> processingEnv.getMessager().printMessage(Diagnostic.Kind.ERROR, error(1013).formatted(method, element.getQualifiedName(), ((TypeElement) overriddenMethod.getEnclosingElement()).getQualifiedName()), method));
```

Comment:
Please refactor this into a lambda with braces { } and perhaps a few variables. Currently the line is way to long.